modelling/src/main/java/org/axonframework/modelling/SimpleRepository.java

```diff
@@ -0,0 +1,195 @@
+ * loaded again in the same context.
+ * <p>
+ * When an entity is successfully loaded, it's scheduled to automatically be persisted when the
+ * {@link ProcessingContext} is committed.
+ * <p>
+ * Both the {@link #load(Object, ProcessingContext)} and {@link #loadOrCreate(Object, ProcessingContext)} use the
+ * {@link SimpleRepositoryEntityLoader} to load the entity. If you wish to create an entity when it does not exist, you can return
+ * a value when you have found none.
+ *
+ * @param <I> The type of the identifier of the entity.
+ * @param <T> The type of the entity.
+ * @author Mitchell Herrijgers
+ * @since 5.0.0
+ */
+public class SimpleRepository<I, T> implements AsyncRepository.LifecycleManagement<I, T> {
+
+    private final ResourceKey<Map<I, CompletableFuture<ManagedEntity<I, T>>>> managedEntitiesKey =
+            ResourceKey.withLabel("SimpleRepository.ManagedEntities");
+
+    private final Class<I> idType;
+    private final Class<T> entityType;
+    private final SimpleRepositoryEntityLoader<I, T> loader;
+    private final SimpleRepositoryEntityPersister<I, T> persister;
+
+    /**
+     * Constructs a new simple {@link AsyncRepository} for entities of type {@code entityType} with identifiers of type
+     * {@code idType}. The given {@code loader} is used to load entities and the given {@code persister} is used
+     * to persist entities.
+     *
+     * @param idType     The type of the identifier of the entity.
+     * @param entityType The type of the entity.
+     * @param loader     The component capable of loading entities.
+     * @param persister  The component capable of persisting entities.
+     */
+    public SimpleRepository(
+            @Nonnull Class<I> idType,
+            @Nonnull Class<T> entityType,
+            @Nonnull SimpleRepositoryEntityLoader<I, T> loader,
+            @Nonnull SimpleRepositoryEntityPersister<I,T> persister
+    ) {
+        this.idType = requireNonNull(idType, "The entityIdClass may not be null");
+        this.entityType = requireNonNull(entityType, "The entityClass may not be null");
+        this.loader = requireNonNull(loader, "The loader may not be null");
+        this.persister = requireNonNull(persister, "The persister may not be null");
+    }
+
+    @Override
+    public ManagedEntity<I, T> attach(@Nonnull ManagedEntity<I, T> entity,
+                                      @Nonnull ProcessingContext context) {
+        var managedEntities = context.computeResourceIfAbsent(managedEntitiesKey, ConcurrentHashMap::new);
+
+        return managedEntities.computeIfAbsent(
+                entity.identifier(),
+                id -> {
+                    SimpleEntity<I, T> simpleEntity = SimpleEntity.mapToSimpleEntity(entity);
+                    attachEntitySaveToContext(id, simpleEntity, context);
+                    return CompletableFuture.completedFuture(simpleEntity);
+                }
+        ).resultNow();
+    }
+
+    @javax.annotation.Nonnull
```

Comment:
I think this should be the jakarta annotation, not the javax one.

Reply:
How did these get here?! Removing